lib/Server.php

```diff
@@ -20,11 +20,16 @@ public function __construct($socket) {
                 \stream_set_blocking($client, false);
                 $promisor = $state->promisor;
                 $state->promisor = new amp\Deferred;
-                $promisor->succeed(new Client($client));
+                $promisor->succeed(static::createClient($client));
```

Comment:
in case we stay with the name `createClient` we should rename the local `client` var.
atm it feels a bit mixed up.